addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -508,6 +508,42 @@ var NumericField = InputField.extend({
         }
         return this._super(value, options);
     },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Replace the decimal separator of the numpad dot key
+     * by the decimal separator from the user's language setting.
+     *
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onKeydown(ev) {
+        const kbdEvt = ev.originalEvent;
+        if (kbdEvt && utils.isNumpadDecimalSeparatorKey(kbdEvt)) {
+            const inputField = this.$input[0];
+            const curVal = inputField.value;
+            const from = inputField.selectionStart;
+            const to = inputField.selectionEnd;
+            const point = _t.database.parameters.decimal_point;
+
+            // Make sure the correct decimal separator
+            // from the user's settings is inserted
+            inputField.value = curVal.slice(0, from) + point + curVal.slice(to);
+
+            // Put the user caret at the right place
+            inputField.selectionStart = inputField.selectionEnd = from + point.length;
+
+            // Prevent event default behaviour but
+            // still tell the world we made some changes.
+            kbdEvt.preventDefault();
+            this._doDebouncedAction();
+        }
```

Comment:
I think we can return here. the super only does navigation stuff, so we know we won't be in this case. Moreover, the default is prevented, suggesting we only want this function to handle the event.
Admittedly, it won't change anything, so i let you decide :D